guest/api/tolerations.go

```diff
@@ -0,0 +1,46 @@
+package api
+
+import v1 "sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/api"
+
+const (
+	// TaintNodeUnschedulable will be added when node becomes unschedulable
+	// and removed when node becomes schedulable.
+	TaintNodeUnschedulable string = "node.kubernetes.io/unschedulable"
+	// Do not allow new pods to schedule onto the node unless they tolerate the taint,
+	// but allow all pods submitted to Kubelet without going through the scheduler
+	// to start, and allow all already-running pods to continue running.
+	// Enforced by the scheduler.
+	TaintEffectNoSchedule string = "NoSchedule"
+	TolerationOpExists    string = "Exists"
+	TolerationOpEqual     string = "Equal"
+)
+
+// ToleratesTaint checks if the toleration tolerates the taint.
+// The matching follows the rules below:
+//
+//  1. Empty toleration.effect means to match all taint effects,
+//     otherwise taint effect must equal to toleration.effect.
+//  2. If toleration.operator is 'Exists', it means to match all taint values.
+//  3. Empty toleration.key means to match all taint keys.
+//     If toleration.key is empty, toleration.operator must be 'Exists';
+//     this combination means to match all taint values and all taint keys.
+func ToleratesTaint(toleration *v1.Toleration, taint *v1.Taint) bool {
```

Comment:
This refers to this function https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/toleration.go#L38-L57.
As the kubernetes folder is sparsely checked out and I guess should not be modified, is there a better approach how to use this function?

Reply:
Yes, kubernetes folder is only for generating the object files. Also, we cannot import the things from kubernetes because of the limitation coming from TinyGo.
So, yes, it's the only option that we almost do a copy like you do here, at least until we move to [Golang SDK](https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/issues/118).